File: java/src/FibSeq.java
import java.util.*;

public class FibSeq {

    private static Map<Integer, Long> data = new HashMap<>();

    public static void main(String[] args) {
        System.out.println("This is my Feb Seq for learning Dynamic Programming..");

        int n = 500;
        long an = getNthNumber(n);
        System.out.println(n + "th number in my Feb Seq  -> " + an);

    }

    private static long getNthNumber(int n) {

        if (data.containsKey(n))
            return data.get(n);

        if (n == 1 || n == 2)
            return 1;

        long x = getNthNumber(n - 1);
        long y = getNthNumber(n - 2);

        long z = x + y;
        data.put(n, z);
        return z;
    }
}

File: python/fibSeq.py

def getNthNumber(n):

    if n in data.keys():
        return data[n];

    if n == 1 or n == 2:
        return 1
    
    a = n-1
    b = n-2

    x = getNthNumber(a)
    y = getNthNumber(b)
    z = x + y
    data[n] = z

    return z


print("This is my Feb Seq for learning Dynamic Programming..")
n = 500

data = dict()
an = getNthNumber(n)
print(str(n) + "th number in my Feb Seq  -> " + str(an))
